app/assets/javascripts/3dbio_viewer/src/webapp/components/molecular-structure/MolecularStructure.tsx

```diff
@@ -519,22 +537,25 @@ async function applySelectionChangesToPlugin(
         highlight(plugin, chains, newSelection, molstarState);
     }
 
-    plugin.visual.reset({ camera: true });
+    if (added.length + removed.length > 0) {
+        plugin.visual.reset({ camera: true });
+    }
 }
 
 async function highlight(
     plugin: PDBeMolstarPlugin,
     chains: Maybe<PdbInfo["chains"]>,
-    selection: Selection,
-    molstarState: MolstarStateRef
+    selection: BaseSelection,
+    molstarState: MolstarStateRef,
+    focus = true
```

Comment:
[no need to change] Boolean arguments are hard to follow. The typical solution is to add them to an options object (here it could be the full hightlight arguments, or just for focus)